pkg/cloudprovider/provider/openstack/provider.go

```diff
@@ -553,7 +556,9 @@ password = "%s"
 domain-name="%s"
 tenant-name = "%s"
 region = "%s"
-`, c.IdentityEndpoint, c.Username, c.Password, c.DomainName, c.TenantName, c.Region)
+`, re.Replace(c.IdentityEndpoint), re.Replace(c.Username), re.Replace(c.Password), re.Replace(c.DomainName),
```

Comment:
Instead of the `strings.Replacer` approach, the `%s` in the fmt string should be changed to `%q`: `a double-quoted string safely escaped with Go syntax` as per https://golang.org/pkg/fmt/